### PR TITLE
Alllow the user to configure when to empty the AUX ringbuffer.

### PR DESCRIPTION
**Pull request by vext01**
I think it would be a good idea to expose this to the user as a configuration variable.

See individual commits.

Thanks

### CODE REVIEW

File: src/backends/perf_pt/mod.rs [hunk 1]
```diff
@@ -416,7 +427,9 @@ impl PerfPTTracer {
         if !power_of_2(config.aux_bufsize) {
             return Err(HWTracerError::BadConfig(String::from("aux_bufsize must be a positive power of 2")));
         }
-
+        if config.aux_copyout_threshold < 0. || config.aux_copyout_threshold > 1. {
```
Comment (ltratt) [inline src/backends/perf_pt/mod.rs]: Shouldn't this error be in the `aux_copyout_threshold` function?

Comment (vext01) [reply src/backends/perf_pt/mod.rs]: That's a copy-pasta, indeed.

**Comment by vext01** [comment]
> I don't think we should bother setting configuration options until we know we'll need them.

I agree that I am speculating the need for this config option. I'm imagine that different parameters will be needed for different hardware (depending upon the speed the CPU execute code, and the speed of the memory subsystem, etc.).

If you still feel that this is unneccessary, I suggest we close this and leave the branch around until we have at-least tested on some different hardware.

Thoughts?

**Comment by ltratt** [comment]
I must admit, I would prefer to do this on an as-needed basis.

**Comment by vext01** [comment]
OK, closing for now.

**Pull request closed without merging**